### PR TITLE
Use `override_limits` from config when present

### LINKED ISSUE #94: Use `override_limits` from config when present

### PR DESCRIPTION
**Pull request by MasloMaslane**
Closes #94

### CODE REVIEW

File: src/sinol_make/helpers/package_util.py [hunk 1]
```diff
@@ -72,15 +76,28 @@ def get_time_limit(test_path, config, args=None):
...
+            return None
+
+    if "override_limits" in str_config and lang in str_config["override_limits"]:
+        limit = get_time_limit_from_dict(str_config["override_limits"][lang])
+        if limit is None:
+            return get_time_limit_from_dict(str_config)
+        else:
+            return limit
+    else:
+        return get_time_limit_from_dict(str_config)
```
Comment (tonowak) [inline src/sinol_make/helpers/package_util.py:97]: Any ideas how to make this code more readable and less prone to bugs?

File: src/sinol_make/helpers/package_util.py [hunk 2]
```diff
@@ -91,9 +108,22 @@ def get_memory_limit(test_path, config, args=None):
...
+            return None
+
+    if "override_limits" in str_config and lang in str_config["override_limits"]:
+        limit = get_memory_limit_from_dict(str_config["override_limits"][lang])
+        if limit is None:
+            return get_memory_limit_from_dict(str_config)
+        else:
+            return limit
+    else:
+        return get_memory_limit_from_dict(str_config)
```
Comment (tonowak) [inline src/sinol_make/helpers/package_util.py:129]: Any idea how to reduce the amount of copy-pasted code?

File: src/sinol_make/helpers/package_util.py [hunk 3]
```diff
@@ -61,39 +62,65 @@ def get_executable_path(solution: str) -> str:
...
+
+def _get_limit_from_dict(dict: Dict[str, Any], limit_type: LimitTypes, test_id: str, test_group: str):
+    if limit_type == LimitTypes.TIME_LIMIT:
+        limit_name = "time_limit"
+        plural_limit_name = "time_limits"
+    elif limit_type == LimitTypes.MEMORY_LIMIT:
+        limit_name = "memory_limit"
+        plural_limit_name = "memory_limits"
+    else:
+        raise ValueError("Invalid limit type.")
```
Comment (tonowak) [inline src/sinol_make/helpers/package_util.py:82]: I believe there should be a `match` statement in Python, which would work on enums?

**Review by tonowak — APPROVED** [review]
Nice!

File: src/sinol_make/helpers/package_util.py (hunk 3 above)
Comment (MasloMaslane) [reply src/sinol_make/helpers/package_util.py:82]: `match` is only supported from python 3.10

**Pull request merged**